src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -509,6 +522,112 @@ public void record(@NonNull SCMHead head, SCMRevision revision, boolean isMatch)
+                                        @SuppressFBWarnings(value = "NP_LOAD_OF_KNOWN_NULL_VALUE",
+                                                            justification =
+                                                                    "TreeWalk.forPath can return null, compiler "
+                                                                            + "generated code for try with resources handles it")
+                                        public SCMProbeStat stat(@NonNull String path) throws IOException {
+                                            try (TreeWalk tw = TreeWalk.forPath(repository, path, tree)) {
+                                                if (tw == null) {
+                                                    return SCMProbeStat.fromType(SCMFile.Type.NONEXISTENT);
+                                                }
+                                                FileMode fileMode = tw.getFileMode(0);
+                                                if (fileMode == FileMode.MISSING) {
+                                                    return SCMProbeStat.fromType(SCMFile.Type.NONEXISTENT);
+                                                }
+                                                if (fileMode == FileMode.EXECUTABLE_FILE) {
+                                                    return SCMProbeStat.fromType(SCMFile.Type.REGULAR_FILE);
+                                                }
+                                                if (fileMode == FileMode.REGULAR_FILE) {
+                                                    return SCMProbeStat.fromType(SCMFile.Type.REGULAR_FILE);
+                                                }
+                                                if (fileMode == FileMode.SYMLINK) {
+                                                    return SCMProbeStat.fromType(SCMFile.Type.LINK);
+                                                }
+                                                if (fileMode == FileMode.TREE) {
+                                                    return SCMProbeStat.fromType(SCMFile.Type.DIRECTORY);
+                                                }
+                                                return SCMProbeStat.fromType(SCMFile.Type.OTHER);
+                                            }
+                                        }
+                                    };
+                                }
+                            }, new SCMSourceRequest.LazyRevisionLambda<GitTagSCMHead, GitTagSCMRevision, ObjectId>() {
+                                @NonNull
+                                @Override
+                                public GitTagSCMRevision create(@NonNull GitTagSCMHead head, @Nullable ObjectId intermediate)
+                                        throws IOException, InterruptedException {
+                                    return new GitTagSCMRevision(head, ref.getValue().name());
```

Comment:
wow we really need `-source 8`

@MarkEWaite BTW [the great majority of people updating this plugin are already on a version of Jenkins requiring Java 8](http://stats.jenkins.io/pluginversions/git.html) :-)